projects/components/src/score-bar/score-bar.component.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  OnChanges,
+  ViewChild
+} from '@angular/core';
+import { Color } from '@hypertrace/common';
+
+@Component({
+  selector: 'ht-score-bar',
+  styleUrls: ['./score-bar.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="score-bar" #scoreBarContainer>
+      <div *ngIf="this.isSingleBarDisplay; else pillScoreBar" class="continuous">
+        <div class="fill" [ngStyle]="{ width: this.fillWidth, backgroundColor: this.fillColor }"></div>
+      </div>
+
+      <ng-template #pillScoreBar>
+        <div class="pills">
+          <div
+            *ngFor="let pill of this.pills; index as index"
+            class="pill"
+            [htTooltip]="pill.value"
+            [ngStyle]="{ backgroundColor: pill.color, width: this.pillWidth }"
+          ></div>
+        </div>
+      </ng-template>
+    </div>
+  `
+})
+export class ScoreBarComponent implements OnChanges, AfterViewInit {
```

Comment:
"Progress bar" would sound better. Any particular reason for the chosen name?

Reply:
@anandtiwary I initially had this as a progress bar, but later changed it since progress bars usually have a slightly different API that includes animation type for non determinate values, etc.

This felt more oriented towards a particular score. Its the same reason why I didn't want to use the material progress bar underneath the hood for the single bar display type. Just leaving space for when we want to actually create a wrapper for the material progress bar.

Reply:
Okay, then should this be part of ht components? May be we put this in the main repo if it wont be shared.